### PR TITLE
docs: updated README.md to include link to repo, files list and description

### LINKED ISSUE #8160: Binary release tarballs should come with a README
A typical release tarball (or zipball) currently contains the following:

```
bitcoin-0.12.1/
bitcoin-0.12.1/bin/
bitcoin-0.12.1/bin/bitcoin-cli
bitcoin-0.12.1/bin/bitcoind
bitcoin-0.12.1/bin/bitcoin-qt
bitcoin-0.12.1/bin/bitcoin-tx
bitcoin-0.12.1/bin/test_bitcoin
bitcoin-0.12.1/include/
bitcoin-0.12.1/include/bitcoinconsensus.h
bitcoin-0.12.1/lib/
bitcoin-0.12.1/lib/libbitcoinconsensus.so
bitcoin-0.12.1/lib/libbitcoinconsensus.so.0
bitcoin-0.12.1/lib/libbitcoinconsensus.so.0.0.0
```

I think we should add a README.md with the release that describes how to get started, and what every file is for.

For windows we have `doc/README_windows.txt` which gets packaged as part of the installer. It is not packaged as part of the windows zips though.

### LINKED ISSUE #8161: Binary release doesn't make it easy to find the source code
This has been suggested before, but I cannot find an issue for it.

I think we should include a link to this repository in various places. At least:
- [x] The "About" dialog in the GUI [done in #8207]
- [ ] A README that comes with release tarballs (see #8160)

This makes it easier to find the source code for potential contributors, as well as avoids awkward questions like #8157.

### PR DESCRIPTION
**Pull request by cisba**
This fixes #8160 (partially), fixes #8161, fixes #11971

### CODE REVIEW

**Review by randolf — APPROVED** [review]

**Comment by laanwj** [comment]
I'm not sure this fixes #8161? Does this add a link to the source repository so that people can find the source code easier?

**Comment by cisba** [comment]
Added the link to repo.

File: doc/README.md [hunk 1]
```diff
@@ -81,4 +98,4 @@ License
 ---------------------
 Distributed under the [MIT software license](/COPYING).
 This product includes software developed by the OpenSSL Project for use in the [OpenSSL Toolkit](https://www.openssl.org/). This product includes
-cryptographic software written by Eric Young ([eay@cryptsoft.com](mailto:eay@cryptsoft.com)), and UPnP software written by Thomas Bernard.
+cryptographic software written by Eric Young ([eay@cryptsoft.com](mailto:eay@cryptsoft.com)), and UPnP software written by Thomas Bernard.
```
Comment (fanquake) [inline doc/README.md]: What is changing here?

Comment (AkioNak) [reply doc/README.md]: It seems that the last '\n' has been deleted.

Comment (cisba) [reply doc/README.md]: Yes, not intentional but also not relevant (IMHO)...

Comment (cisba) [reply doc/README.md]: @fanquake I restored the last '\n' and now it seem like I added a blank line... LOL

**Comment by hebasto** [comment]
> For the sake of style consistency all executables names should be either quoted within the backticks or not quoted at all.

I meant `bitcoin-qt.exe` in the Running Windows section as well.

File: doc/README.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ Unpack the files into a directory and run:
 
 ### Windows
 
-Unpack the files into a directory, and then run
```
Comment (cisba) [inline doc/README.md]: trailing whitespace generated error in `test/lint/lint-whitespace.sh`
https://travis-ci.org/bitcoin/bitcoin/jobs/425139871

File: doc/README.md (hunk 1 above)
Comment (cisba) [reply doc/README.md]: fixed

**Comment by cisba** [comment]
@laanwj I think all points now are checked. Anyway, I made a new PR #14167 with the aim to improve even more the documentation in tar packages.

**Comment by merland** [comment]
In [PR 14755](https://github.com/bitcoin/bitcoin/pull/14755), I suggest to remove all instructions about running and instead link to the official docs, to avoid duplication. Currently in conflict with this PR. Review/comment requested.  
(Also related: #14639)

**Comment by fanquake** [comment]
Closing this, given that #14167 includes these changes.

**Pull request closed without merging**